allennlp/models/ensemble/ensemble.py

```diff
@@ -15,7 +16,29 @@ def __init__(self,
 
         super(Ensemble, self).__init__(vocab, None)
         self.submodels = submodels
-        self.is_ensemble = True
 
     def forward(self, *inputs) -> Dict[str, torch.Tensor]:  # pylint: disable=arguments-differ
         raise NotImplementedError
+
+    @classmethod
+    def _load(cls, config: Params, serialization_dir: str, weights_file: str = None, cuda_device: int = -1) -> 'Model':
+        """
+        Ensembles don't have vocabularies or weights, so they override _load.
+        """
+        model_params = config.get('model')
+
+        # The experiment config tells us how to _train_ a model, including where to get pre-trained
+        # embeddings from.  We're now _loading_ the model, so those embeddings will already be
+        # stored in our weights.  We don't need any pretrained weight file anymore, and we don't
+        # want the code to look for it, so we remove it from the parameters here.
+        # Model._remove_pretrained_embedding_params(model_params) TODO(michaels)
+        model = Model.from_params(None, model_params)
```

Comment:
This basic structure looks much better!  I'm guessing that you can just iterate over some `submodels` key in the config, and call `Model.load()` or `load_archive()`, or something like that.

Reply:
I don't understand your point.  This is exactly what happens in [https://github.com/allenai/allennlp/pull/1000/files#diff-237752509d4dbe981cfb0c60400a0f8dR105](BidafEnsemble.from_params).

Reply:
Ah, sorry, I didn't realize you were doing that logic there.  Yeah, that's probably ok.